src/realm/util/features.h

```diff
@@ -292,6 +292,12 @@
 #define REALM_ARCHITECTURE_X86_64 0
 #endif
 
+#if defined REALM_ARCHITECTURE_X86_32 && defined REALM_WINDOWS
```

Comment:
For these kind of things, I always like to include a link to the bug report so that we know what it is for. Should make it easier to remove it in the future if we forget to and someone stumbles across it and wonders what it is for. `// Temporary workaround for https://developercommunity.visualstudio.com/content/problem/994075/64-bit-atomic-load-ices-cl-1924-with-o2-ob1.html`

I don't love that the macro name is a bit misleading, but it also doesn't matter much since it should be short lived anyway. They've fixed it in their internal master repo, so it should be included in an msvc update within a few weeks.

Reply:
I realized that I could limit the workaround to alloc.hpp.